packages/runtime/src/runtime.ts

```diff
@@ -259,6 +281,36 @@ export class Runtime {
     }
   }
 
+  /**
+   * Creates Asset in Runtime
+   * @param name ASA name
+   * @param flags ASA Deployment Flags
+   */
+  createAsset (name: string, flags: ASADeploymentFlags): number {
+    const sender = flags.creator;
+    const senderAcc = this.assertAccountDefined(sender.addr, this.store.accounts.get(sender.addr));
+
+    // create asset
+    senderAcc.createAsset(++this.assetCounter, name, this.loadedAssetsDefs[name], sender.addr);
+    this.store.assetDefs.set(this.assetCounter, sender.addr);
+    return this.assetCounter;
+  }
+
+  /**
+   * Returns Asset Definitions
+   * @param assetId Asset Index
+   */
+  getAssetDef (assetId: number): AssetDef {
+    const addr = this.store.assetDefs.get(assetId);
+    if (addr === undefined) {
+      throw new TealError(ERRORS.ASA.ASSET_NOT_FOUND, { assetId: assetId });
+    }
```

Comment:
We should use `this.assertAddressDefined` here (which throws the correct error). Address is undefined here, not the asset.

Reply:
if assetId doesn't exist in map, what does it mean ?

Reply:
hmm, right. Possible that the account & address exists but asset in account doesn't. Then we should update `getApp` as well, as it's using `assertAddressDefined` with globalApps.